languages/rust/src/lib.rs

```diff
@@ -3,213 +3,47 @@
-            ModuleFFI::User(ref user) => &user.module_meta,
-        }
-    }
-}
-
-/// Standard Modules are modules that are built into Flipper and always
-/// available on the device.
-///
-/// Standard Modules must implement `new` and `bind`. The `new` constructor
-/// gives an instance of the module that is attached to the "default"
-/// Flipper. `bind` gives an instance which is attached to a specific
-/// Flipper.
-pub trait StandardModule {
-    fn new() -> Self;
-    fn bind(flipper: &Flipper) -> Self;
-}
-
-/// User Modules are modules which are written in C by the user.
-///
-/// When a user creates a module, they give it a name which is used by
-/// Flipper to load and bind to it. To use it from the Rust bindings,
-/// the user must specify the name so that rust can find the module.
-pub trait UserModule<'a>: From<(&'a Flipper, UserModuleFFI)> {
-    const NAME: &'a str;
-
-    /// Binds an instance of a User Module to the given Flipper.
-    ///
-    /// ```
-    /// use flipper::{Flipper, UserModule, ModuleFFI, UserModuleFFI};
-    ///
-    /// struct MyModule<'a> {
-    ///     flipper: &'a Flipper,
-    ///     module: ModuleFFI,
-    /// }
-    ///
-    /// impl<'a> UserModule<'a> for MyModule<'a> {
-    ///     const NAME: &'a str = "My module";
-    ///     fn new(flipper: &'a Flipper) -> Self {
-    ///         MyModule {
-    ///             flipper,
-    ///             module: ModuleFFI::User(UserModuleFFI::uninitialized(Self::NAME)),
-    ///         }
-    ///     }
-    /// }
-    ///
-    /// impl<'a> From<(&'a Flipper, UserModuleFFI)> for MyModule<'a> {
-    ///     fn from((flipper, module): (&'a Flipper, UserModuleFFI)) -> Self {
-    ///         MyModule {
-    ///             flipper,
-    ///             module: ModuleFFI::User(module),
-    ///         }
-    ///     }
-    /// }
-    ///
-    /// impl<'a> MyModule<'a> {
-    ///     fn myFunc(&self) {
-    ///         // Do FMR invocation
-    ///     }
-    /// }
-    ///
-    /// let flipper = Flipper::attach();
-    /// let myModule = MyModule::new(&flipper); // Binds "MyModule" to control "flipper".
-    ///
-    /// myModule.myFunc();
-    /// ```
-    fn new(flipper: &'a Flipper) -> Self {
-        let mut module = UserModuleFFI::uninitialized(Self::NAME);
-        unsafe { lf_bind(flipper.device, &mut module.module_meta); }
-        Self::from((flipper, module))
-    }
-}
-
 #[link(name = "flipper")]
 extern {
     fn flipper_attach() -> _lf_device;
     fn carbon_attach_hostname(hostname: *const c_char) -> _lf_device;
-    fn lf_bind(device: *const c_void, module: *mut _lf_module) -> c_int;
+    fn lf_get_current_device() -> _lf_device;
 }
 
 pub struct Flipper {
     /// A reference to an active Flipper profile in libflipper. This
     /// is used when communicating with libflipper to specify which
     /// device functions should be executed on.
-    device: _lf_device,
+    device: _lf_device
```

Comment:
I tend to like keeping the trailing commas on multi line lists like structs. If you remember to set this back that'd be cool.